### ISSUE TITLE
C++ is great, but mass renaming is a quite awkward.

### ISSUE THREAD

**Issue by skull-squadron** [myang-git/QR-Code-Encoder-for-Objective-C#25]
I love C++11 meta awesome as much as the next guy, but mass renaming, especially for existing apps, is painful configuration over convention.

I audited the class, it's not really doing anything C++ like except hiding some private instance vars that get passed around.  And there's only one public method, so it seems kind of goofy to use classes.   I'll demonstrate a C++ -> C trick of how to hide private instance data w/ public in 1 malloc.

Having it as a C99 QR library is also useful.

Legacy code-style transition:
1. Keep the existing classes that proxy to C equivalents (no mm -> m required).
2. Simple #ifdef to choose C or C++ version in the one place w/ new and delete.
3. Optional deprecations notices later.

--excuses for not using this awesome library! :)

**Comment by myang-git** [comment]
You are right. The C++ code here is not doing real C++ stuffs. It's only the matter of time and some testing for porting it to C. I may spend a weekend or two on this later.

**Comment by skull-squadron** [comment]
Beat you to it, my bad. :)

Translated Japanese -> English comments too, the lazy Google way.

**Comment by skull-squadron** [comment]
The QR encoder itself: It could use lots of work refactoring common non-DRY parts.  It's much too long and repetitious.  Also, the functions are much too big / complicated to be testable / maintainable.

**Comment by skull-squadron** [comment]
Alright so the (Objective|)C(|++) is at a working point.  Feel free to hack on this, throw it away, whatever.
